### ISSUE TITLE
Trimming both sides of an if/else

### ISSUE THREAD

**Issue by jbt** [rollup/rollup#1113]
I'm seeing what looks like both sides of an if/else getting removed if there's a `var` inside with rollup 0.36.3

Here's an example:

```js
if (foo) {
  var x = 1;
} else {
  var x = 2;
}
console.log(x);
```

This apparently gets transformed to just `console.log(x)` and `x` is never declared (example [here](http://rollupjs.org/?version=0.36.3&shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybW9kdWxlTmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMm1vZHVsZUlkJTIyJTNBJTIyJTIyJTdEJTJDJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmlmJTIwKGZvbyklMjAlN0IlNUNuJTIwJTIwdmFyJTIweCUyMCUzRCUyMDElM0IlMjAlNUNuJTdEJTIwZWxzZSUyMCU3QiU1Q24lMjAlMjB2YXIlMjB4JTIwJTNEJTIwMiUzQiU1Q24lN0QlNUNuJTVDbmNvbnNvbGUubG9nKHgpJTNCJTIyJTdEJTVEJTdE))

Seems to work fine if the `foo` is replaced by various different things like `true` or `foo()`

Slightly less contrived example: [this](https://github.com/facebook/react/blob/1c1f68e8dccbc84acfae66178493dbfb8ca7d1e1/src/isomorphic/hooks/ReactComponentTreeHook.js#L65) whole if/else in React 15.4.0 (just published) is getting stripped out, so `getRootIDs` used [later on](https://github.com/facebook/react/blob/bdf263625db59f54368d39e8ef077e7f0e9313db/src/isomorphic/hooks/ReactComponentTreeHook.js#L402) is throwing a ReferenceError.

Happy to provide any further examples if needed.

**Comment by btd** [comment]
To add some clarity. If we take such code:
```js
var a = true;
if(a) { // IfStatement will have testValue === UNKNOWN
  var b = 10;
} else {
  var b = 11;
}
console.log(b);
```
First stripped VariableDeclaration `var a = true`, next whole IfStatement disappeared. If we move `var b;` outside of `if`, rollup can successfully rest if in place.

**Comment by samjonester** [comment]
Is variable hoisting no longer a part of the ECMAScript spec? Regardless, I would imagine this will push many users from rollup if it does not work with react because hoisting is not supported. I also doubt react is the only popular library that uses hoisting in this manner. Side note: react seems to think this is a problem with rollup: https://github.com/facebook/react/issues/8318

**Comment by rgamm** [comment]
I came across this exact issue and had to downgrade react to fix. However I've since attempted to use another 3rd party package (react-datepicker) which has some code in it's dist file which breaks when put through rollup.

Essentially their bundle code seems to be generated by webpack and ends up looking like:
```javascript
function extend() {} // this gets pruned
SomeObject = { extend: extend }
var extend = SomeObject.extend;
var broken = extend(a, b);
```
[live rollup example](http://rollupjs.org/?version=0.36.3&shareable=JTdCJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmFtZCUyMiUyQyUyMm1vZHVsZU5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJtb2R1bGVJZCUyMiUzQSUyMiUyMiU3RCUyQyUyMm1vZHVsZXMlMjIlM0ElNUIlN0IlMjJuYW1lJTIyJTNBJTIybWFpbi5qcyUyMiUyQyUyMmNvZGUlMjIlM0ElMjJmdW5jdGlvbiUyMGV4dGVuZCgpJTIwJTdCJTdEJTVDbnZhciUyMFNvbWVPYmplY3QlMjAlM0QlMjAlN0IlMjBleHRlbmQlM0ElMjBleHRlbmQlMjAlN0QlNUNudmFyJTIwZXh0ZW5kJTIwJTNEJTIwU29tZU9iamVjdC5leHRlbmQlM0IlNUNudmFyJTIwYnJva2VuJTIwJTNEJTIwZXh0ZW5kKGElMkMlMjBiKSUzQiU1Q24lNUNuY29uc29sZS5sb2coZXh0ZW5kKCU3QmElM0ExJTdEJTJDJTIwJTdCYiUzQTIlN0QpKSUzQiUyMiU3RCU1RCU3RA==) which shows you end up with a lot of undefined being passed around.
I'm not 100% sure this is the same thing as the if branches, but seems to be because of hoisting. I am having to move off of rollup until this is resolved though. Is there a way to disable treeshaking on specific files/modules perhaps?

As an aside, I'm also seeing https://github.com/rollup/rollup/issues/1102 when trying to disable the treeshaking, so can't do it for everything.

**Comment by LinusU** [comment]
Also hitting this issue, for anyone coming from google, this is the exact error message: `ReferenceError: Can't find variable: getRootIDs`

**Comment by LinusU** [comment]
This should probably be fixed here as well, but here is a PR with a fix in React: https://github.com/facebook/react/pull/8380

**Comment by MatthewEdge** [comment]
Any updates on this fix? I see a PR got merged in facebook/react#8380 for 15.4.1 but I'm still getting the same issue with getRootIDs with that version

**Comment by Rich-Harris** [comment]
Fixed in 0.37.2 – thanks
